src/main/java/wolox/training/utils/CustomAuthenticationProvider.java

```diff
@@ -0,0 +1,29 @@
+package wolox.training.utils;
+
+import java.util.ArrayList;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CustomAuthenticationProvider implements
+    AuthenticationProvider {
+
+  @Override
+  public Authentication authenticate(Authentication authentication)
+      throws AuthenticationException {
+
+    String name = authentication.getName();
+    String password = authentication.getCredentials().toString();
+
+    return new UsernamePasswordAuthenticationToken(
```

Comment:
You don't check if the password is the correct one.  Add this: 
`@Autowired
    private PasswordEncoder passwordEncoder;`